src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -144,17 +160,18 @@ public HttpRequestFactory createRequestFactory() {
    * @throws IOException if something goes wrong while fetching the token
    */
   public String fetchAccessToken() throws IOException {
```

Comment:
A motivating example, one of the things I loved about Parse was it's background auto-deploy on change.  It was great for development.  With IAM I'd ensure that my production instance couldn't be deployed to by my normal account.

Having to switch the active user on the login state to perform an operation, and then switch back, is not thread safe.